docs/build-docs.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+rm -f ./_FIRST_RUN.flag
+
+ARCH=$(uname -m)
+export PATH="${CONDA}/bin:${PATH}"
+
+curl \
+    -L \
+    -o ${HOME}/conda.sh \
+    "https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-${ARCH}.sh"
+
+/bin/bash ${HOME}/conda.sh -b -p ${CONDA}
+
+conda config --set always_yes yes --set changeps1 no
+
+mamba create \
+    --yes \
+    --name docs-env \
+    --file env.yml || exit -1
+
+source activate docs-env
+
+${CONDA}/envs/docs-env/bin/python \
+    -m sphinx \
+    -T \
+    -E \
+    -W \
+    --keep-going \
+    -b html \
+    -d _build/doctrees \
+    -D language=en \
+    . _build/html || exit -1
```

Comment:
Why do we need this long command and cannot use `make html`?
https://github.com/microsoft/LightGBM/blob/master/docs/Makefile

Reply:
I got this from the RTD builds, e.g. https://readthedocs.org/projects/lightgbm/builds/15824338/. Was trying to mimic exactly what is run in that environment.

<img width="1154" alt="Screen Shot 2022-01-19 at 10 39 14 PM" src="https://user-images.githubusercontent.com/7608904/150274687-1d8b5ab3-a725-4134-8dcc-b60e5aeeffe1.png">

If you think that's not necessary and is the same as running `make html` I'd be happy to change it to `make html`.

Reply:
I think `make html` should be enough. If we really want to mimic `readthedocs`, we should update `make` files, but not duplicate code.
https://github.com/microsoft/LightGBM/blob/master/docs/Makefile
https://github.com/microsoft/LightGBM/blob/master/docs/make.bat